src/impl.cpp

```diff
@@ -0,0 +1,15 @@
+#include "impl.h"
+
+int BinaryInvert(int x,int p,int n)
+{
+    int tmp = 0;
+
+    for(int i = 0; i < n; i++)
```

Comment:
wouldn't it be easier to handle without the loop?
for bitmask preparation you need just two shifts